src/Pos/Util.hs

```diff
@@ -260,8 +260,7 @@ clearMVar = liftIO . void . tryTakeMVar
 
 forcePutMVar :: MonadIO m => MVar a -> a -> m ()
 forcePutMVar mvar val = do
-    res <- liftIO $ tryPutMVar mvar val
-    unless res $ do
+    unlessM (liftIO $ tryPutMVar mvar val) $ do
```

Comment:
You don't need `liftIO` here because `universum` already contains lifted versions.